src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/Extensions/KvsActorStateProviderExtensions.cs

```diff
@@ -198,6 +187,52 @@ internal static bool GetDisableTombstoneCleanupSetting(this KvsActorStateProvide
             return stateProvider.GetStoreReplica().KeyValueStoreReplicaSettings.DisableTombstoneCleanup;
         }
 
+        internal static Task GetValueByKeysAsync(this KvsActorStateProvider stateProvider, List<string> keys, HttpResponse response, CancellationToken cancellationToken)
+        {
+            var pairs = new List<KeyValuePair>();
+            return stateProvider.GetActorStateProviderHelper().ExecuteWithRetriesAsync(
+                        async () =>
+                        {
+                            if (keys.Any())
+                            {
+                                using var tx = stateProvider.GetStoreReplica().CreateTransaction();
+
+                                foreach (var key in keys)
+                                {
+                                    var result = stateProvider.GetStoreReplica().TryGet(tx, key);
```

Comment:
The key should always be found. If the key is not found, then it a failed migration scenario.

Reply:
Added check and throwing exception if key was not found